src/test/java/com/thangle/api/user/UserControllerTest.java

```diff
@@ -0,0 +1,128 @@
+        final var users = buildUsers();
+
+        when(userService.findAll()).thenReturn(users);
+
+        mvc.perform(MockMvcRequestBuilders.get(BASE_URL))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.length()").value(users.size()))
+                .andExpect(jsonPath("$[0].id").value(users.get(0).getId().toString()))
+                .andExpect(jsonPath("$[0].username").value(users.get(0).getUsername()))
+                .andExpect(jsonPath("$[0].firstName").value(users.get(0).getFirstName()))
+                .andExpect(jsonPath("$[0].lastName").value(users.get(0).getLastName()))
+                .andExpect(jsonPath("$[0].avatar").value(users.get(0).getAvatar()))
+                .andExpect(jsonPath("$[0].roleId").value(users.get(0).getRoleId().toString()));
+
+        verify(userService).findAll();
+    }
+
+    @Test
+    void shouldFindById_OK() throws Exception {
+        final var user = buildUser();
+
+        when(userService.findUserById(user.getId())).thenReturn(user);
+
+        mvc.perform(MockMvcRequestBuilders.get(BASE_URL + "/" + user.getId()))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(user.getId().toString()))
+                .andExpect(jsonPath("$.username").value(user.getUsername()))
+                .andExpect(jsonPath("$.firstName").value(user.getFirstName()))
+                .andExpect(jsonPath("$.lastName").value(user.getLastName()))
+                .andExpect(jsonPath("$.avatar").value(user.getAvatar()))
+                .andExpect(jsonPath("$.roleId").value(user.getRoleId().toString()));
+
+        verify(userService).findUserById(user.getId());
+    }
+
+    @Test
+    void shouldCreateUser_OK() throws Exception{
+        final var user = buildUser();
+
+        when(userService.createUser(any(User.class))).thenReturn(user);
+
+        mvc.perform(MockMvcRequestBuilders.post(BASE_URL)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(new ObjectMapper().writeValueAsString(user)))
+                .andExpect(jsonPath("$.id").value(user.getId().toString()))
+                .andExpect(jsonPath("$.username").value(user.getUsername()))
+                .andExpect(jsonPath("$.firstName").value(user.getFirstName()))
+                .andExpect(jsonPath("$.lastName").value(user.getLastName()))
+                .andExpect(jsonPath("$.avatar").value(user.getAvatar()))
+                .andExpect(jsonPath("$.enabled").value(user.getEnabled()))
+                .andExpect(jsonPath("$.roleId").value(user.getRoleId().toString()));
+
+    }
+
+    @Test
+    void shouldUpdateUser_OK() throws Exception {
+        final var userNeedsToBeUpdated = buildUser();
+        final var updatedUser = buildUser();
+        updatedUser.setId(userNeedsToBeUpdated.getId());
+
+        when(userService.updateUser(any(UUID.class), any(User.class))).thenReturn(updatedUser);
```

Comment:
```suggestion
        when(userService.updateUser(any(), any())).thenReturn(updatedUser);
```

When the overload issue doesn't happens, no need to pass the class name